src/Metadata/MethodMetadata.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PHPSharkTank\Anonymizer\Metadata;
+
+class MethodMetadata
+{
+    private $reflection;
+
+    private $name;
+
+    private $className;
+
+    private $annotationClass;
+
+    public function __construct(string $class, string $methodName, string $annotationClass)
+    {
+        $this->className = $class;
+        $this->name = $methodName;
+        $this->annotationClass = $annotationClass;
```

Comment:
Why we need an annotation class? This will brings a dependency between the loader and the metadata. Maybe in some near future, we want to support YAML or XML, too. This won't be possible anymore with this PR.

Reply:
Maybe a little misspelled, or misunderstandable. It contains the class of the event which is bound to the function. So for YAML or XML context it might be a bit confusing.
You also can assume that YAML or XML config adds virtual annotations, so therefore it's an annotation class.
Later it might be possible to use another annotation for another context, so if its renamed to `EventClass` the property name is to special and bound to a specific use case.